plugins/janus_videoroom.c

```diff
@@ -990,6 +990,7 @@ void janus_videoroom_destroy_session(janus_plugin_session *handle, int *error) {
 	if(!g_atomic_int_get(&session->destroyed)) {
 		/* Any related WebRTC PeerConnection is not available anymore either */
 		janus_videoroom_hangup_media(handle);
+		handle->plugin_handle = NULL;
```

Comment:
Instead of setting `handle->plugin_handle` to `NULL` here, I would recommend doing it in `janus_videoroom_session_free`:

	session->handle->plugin_handle = NULL;

That way, the plugin has access to the session until the refcount is zero.